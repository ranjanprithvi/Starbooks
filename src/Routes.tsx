import React from "react";
import { Navigate, Route, Routes as ReactRoutes } from "react-router-dom";
import Books from "./components/Books/Books";
import BookDetails from "./components/Books/BookDetails";
import BookForm from "./components/Books/BookForm";
import NotFound from "./components/common/NotFound";
import Rentals from "./components/Rentals/Rentals";
import Users from "./components/Users/Users";
import RentalForm from "./components/Rentals/RentalForm";
import UserDetails from "./components/Users/UserDetails";
import UserForm from "./components/Users/UserForm";
import LoginForm from "./components/Users/LoginForm";
import ProtectedAdminComponent from "./components/common/ProtectedAdminComponent";
import ProtectedComponent from "./components/common/ProtectedComponent";
import AuthorGenreList from "./components/Authors/AuthorGenreList";
import Home from "./components/Home";

const Routes = ({ dataView }: { dataView?: string | undefined }) => {
    return (
        <ReactRoutes>
            <Route path="/" element={<Home />} />
            <Route path="/login" element={<LoginForm />} />
            <Route path="/books" element={<Books />} />
            <Route path="bookDetails/:id" element={<BookDetails />} />
            <Route
                path="books/:id"
                element={
                    <ProtectedAdminComponent>
                        <BookForm />
                    </ProtectedAdminComponent>
                }
            />
            <Route
                path="users"
                element={
                    <ProtectedAdminComponent>
                        <Users dataView={dataView} />
                    </ProtectedAdminComponent>
                }
            />
            <Route
                path="userDetails/:id"
                element={
                    <ProtectedComponent>
                        <UserDetails />
                    </ProtectedComponent>
                }
            />
            <Route
                path="users/:id"
                element={
                    <ProtectedAdminComponent>
                        <UserForm />
                    </ProtectedAdminComponent>
                }
            />
            <Route
                path="rentals"
                element={
                    <ProtectedAdminComponent>
                        <Rentals />
                    </ProtectedAdminComponent>
                }
            />
            <Route
                path="rentals/:id"
                element={
                    <ProtectedAdminComponent>
                        <RentalForm />
                    </ProtectedAdminComponent>
                }
            />
            <Route path="/authorGenreList" element={<AuthorGenreList />} />
            <Route path="/not-found" element={<NotFound />} />
            <Route path="*" element={<Navigate to="/not-found" replace />} />
        </ReactRoutes>
    );
};

export default Routes;
