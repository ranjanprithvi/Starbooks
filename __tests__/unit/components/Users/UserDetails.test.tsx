import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";

import Routes from "../../../../src/Routes";
import { MemoryRouter } from "react-router-dom";

//mock useUser
jest.mock("../../../../src/hooks/useUser");

jest.mock("../../../../src/components/common/ProtectedComponent");

describe("UserDetails ", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(
                <MemoryRouter initialEntries={[`/userDetails/123`]}>
                    <Routes />
                </MemoryRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders error message when no user with given Id is found", () => {
        const errorMsg = "User Not Found!";

        render(
            <MemoryRouter initialEntries={[`/userDetails/invalid`]}>
                <Routes />
            </MemoryRouter>
        );

        expect(screen.getByText(errorMsg)).toBeInTheDocument();
    });
});
