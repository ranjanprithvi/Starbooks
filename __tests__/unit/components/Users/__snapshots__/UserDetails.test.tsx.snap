// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails  renders correctly 1`] = `
<div
  className="css-1foy0fl"
>
  <div
    className="css-t517ds"
  >
    <div
      className="css-1v81ncp"
    >
      <div
        className="css-1gvp5h0"
      >
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-1upb9tn"
        />
        <div
          className="css-8g2ide"
        >
          <div
            className="css-azx95j"
          >
            <h2
              className="chakra-heading css-0"
            >
              test
            </h2>
          </div>
        </div>
      </div>
      <div
        className="css-1gvp5h0"
      >
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-1upb9tn"
        />
        <div
          className="css-8hm5fo"
        >
          <p
            className="chakra-text css-in3yi3"
          >
            Email 
          </p>
           
          <p
            className="chakra-text css-0"
          >
            user@starbooks.com
          </p>
        </div>
      </div>
      <div
        className="css-1gvp5h0"
      >
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-1upb9tn"
        />
        <div
          className="css-8hm5fo"
        >
          <p
            className="chakra-text css-in3yi3"
          >
            Phone Number
          </p>
           
          <p
            className="chakra-text css-0"
          >
            +49 1234567890
          </p>
        </div>
      </div>
      <div
        className="css-1gvp5h0"
      >
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-1upb9tn"
        />
        <div
          className="css-8hm5fo"
        >
          <p
            className="chakra-text css-in3yi3"
          >
            Date of Birth: 
          </p>
           
          <p
            className="chakra-text css-0"
          >
            01 Jan 2000
          </p>
        </div>
      </div>
      <div
        className="css-1gvp5h0"
      >
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-1upb9tn"
        />
        <div
          className="css-8hm5fo"
        >
          <p
            className="chakra-text css-in3yi3"
          >
            Membership Expiry: 
          </p>
           
          <p
            className="chakra-text css-bjcoli"
          >
            01 Jan 2031
          </p>
        </div>
      </div>
      <div
        className="css-1gvp5h0"
      >
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-1upb9tn"
        />
        <div
          className="css-8hm5fo"
        >
          <p
            className="chakra-text css-in3yi3"
          >
            Book Limit: 
          </p>
           
          <p
            className="chakra-text css-0"
          >
            2
          </p>
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-1upb9tn"
      />
      <div
        className="css-ay71wn"
      >
        <div
          className="chakra-stack css-1y3f6ad"
        >
          <h2
            className="chakra-heading css-0"
          >
            Active Rentals
          </h2>
        </div>
        <hr
          aria-orientation="horizontal"
          className="chakra-divider css-9c8gsq"
        />
        <p
          className="chakra-text css-0"
        >
          There are currently no active rentals
        </p>
      </div>
    </div>
  </div>
</div>
`;
